gammapy/stats/variability.py

```diff
@@ -228,3 +230,101 @@ def compute_flux_doubling(flux, flux_err, coords, axis=0):
     }
 
     return doubling_dict
+
+
+def TimmerKonig_lightcurve_simulator(
+    power_spectrum,
+    npoints,
+    spacing,
+    function="powerlaw",
+    random_state="random-seed",
+    normalization=1.0,
+):
+    """Implementation of the Timmer-Koenig algorithm to simulate a time series from a power spectrum.
+
+    Parameters
+    ----------
+    power_spectrum : float, function
+        Power spectrum used to generate the time series, presented analytically, as a parameter or set
+        of parameters of a function. If the type parameter is set to "custom", power_spectrum is expected to be
+        a function mapping the input frequencies to the periodogram.
+    npoints : float
+        Number of points in the output time series.
+    spacing : float
+        Sample spacing, inverse of the sampling rate.
+    function : {"powerlaw", "white", "custom"}
+        Type of input periodogram.  Default is "powerlaw".
+        If 'type' = "powerlaw", 'powerspectrum' is expected to be  a float representing the power index.
+        If 'type' = "white", 'powerspectrum' is expected to be  a float representing the standard deviation.
+        If 'type' = "custom", 'powerspectrum is expected to be a function of the input frequencies.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+        Defines random number generator initialisation.
+        Passed to `~gammapy.utils.random.get_random_state`. Default is "random-seed".
+    normalization : float
+        Normalization factor to be applied to the final time series
+
+    Returns
+    -------
+    time_series: `~numpy.ndarray`
+        Simulated time series.
+
+    References
+    ----------
+    ..[Timmer1995]"On generating power law noise", J. Timmer and M, Konig, section 3.
+    """
+    if function == "custom" and not isinstance(power_spectrum, type(lambda: None)):
+        raise ValueError("A custom power spectrum has to be provided as a function.")
+
+    random_state = get_random_state(random_state)
+
+    frequencies = np.fft.fftfreq(npoints, spacing)
+
+    # To obtain real data only the positive or negative part of the frequency is necessary.
+    real_frequencies = np.sort(np.abs(frequencies[frequencies < 0]))
+
+    if function == "powerlaw":
+        periodogram = (1 / real_frequencies) ** power_spectrum
+    elif function == "white":
+        periodogram = np.full_like(real_frequencies, power_spectrum)
+    elif function == "custom":
+        periodogram = power_spectrum(real_frequencies)
+    else:
+        raise ValueError("Power spectrum function not accepted!")
```

Comment:
I think it is simpler/safer to accept only a callable and explain in the docstring that to generate a PL noise, one needs to pass `power_spectrum = lambda f : f**-index` or `power_spectrum = lambda f : sigma`